packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -196,14 +233,19 @@ export abstract class TelemetryLogger implements ITelemetryLoggerExt {
 	 *
 	 * @param event - Event to send
 	 * @param error - optional error object to log
+	 * @param logLevel - optional level of the log.
 	 */
-	public sendPerformanceEvent(event: ITelemetryPerformanceEventExt, error?: any): void {
+	public sendPerformanceEvent(
+		event: ITelemetryPerformanceEventExt,
+		error?: any,
+		logLevel?: LogLevel.verbose | LogLevel.default,
```

Comment:
Just like this only takes verbose and default, I think `sendTelemetryEvent()` should only take `verbose, default, warning`, instead of us magically changing a potential `error`. Feels like we can also do `this.sendTelemetryEventCore(event, error, logLevel ?? LogLevel.default)` both here and in `sendTelemetryEvent()`, and get rid of `decideLogLevel`.

Reply:
I see the current version of the PR went the other way. With the removal of warning, I think there's even more reason to restrict `sendTelemetryEvent` and `sendPerformanceEvent` to only take the values that we'll preserve. Thoughts?

Reply:
So both the perf and generic event can be sent with category as "error" in case something happens and the sendTelemetryEvent api needs to be of category error. In that case, if the caller calls sendTelemetryEvent with category as "error" and log level as default, then we should internally change that log level to "error".

Reply:
Also added tests and removed warning for now, as it can always be added later on if need be.

Reply:
That (overriding logLevel internally based on category) seems ok but should be documented as part of the `@remarks` for both methods. "If the category of the event is error, the logLevel will be set to error too." But accepting an event with category "generic" and logLevel "error", and silently changing the logLevel to be default seems weird. It could even cause events to stop flowing to the base logger when arguably they should (e.g. the base logger wants minLevel error, so they would have gotten that event if we didn't override the logLevel with default).

I still think a cleaner approach would be that the parameter in both cases only accepts `LogLevel.default` and `LogLevel.verbose` (we could expand that later if necessary without it being breaking), we use `LogLevel.default` as default value for the parameter in both cases, and the only override we do is changing it to error when the event category is error.